packages/jest-cli/src/lib/setWatchMode.js

```diff
@@ -25,10 +23,17 @@ const setWatchMode = (
   }
 
   // Reset before setting these to the new values
-  argv._ = (options && options.pattern) || '';
   argv.onlyChanged = false;
-  argv.onlyChanged =
-    buildTestPathPatternInfo(argv).input === '' && !argv.watchAll;
+  delete argv.testPathPattern;
+  delete argv.testNamePattern;
+
+  if (options && options.onlyChanged) {
```

Comment:
I think we can use a the default param syntax to prevent all of the `options &&`